src/Passes/.clang-tidy

```diff
@@ -0,0 +1,76 @@
+Checks: "-*,bugprone-*,\
+-readability-*,\
+readability-identifier-*,\
+readability-braces-around-statements,\
+cert-dcl*,\
+cert-env*,\
+cert-err52-cpp,\
+cert-err60-cpp,\
+cert-flp30-c,\
+clang-analyzer-security.FloatLoopCounter,\
+google-build-explicit-make-pair,\
+google-build-namespaces,\
+google-explicit-constructor,\
+google-readability-*,\
+google-runtime-operator,\
+hicpp-exception-baseclass,\
+hicpp-explicit-conversions,\
+hicpp-use-*,\
+misc-*,\
+-misc-misplaced-widening-cast,\
+performance-*"
+
+WarningsAsErrors: '*'
+HeaderFilterRegex: '.*'
+
+CheckOptions:
+  - key:    readability-identifier-naming.ClassCase
+    value:  'CamelCase'
+  - key:    readability-identifier-naming.AbstractClassPrefix
+    value:  'I'
+  - key:    readability-identifier-naming.StructCase
+    value:  'CamelCase'
+  - key:    readability-identifier-naming.ParameterCase
+    value:  'lower_case'
+  - key:    readability-identifier-naming.PrivateMemberCase
+    value:  'lower_case'
+  - key:    readability-identifier-naming.PrivateMemberSuffix
+    value:  '_'    
+  - key:    readability-identifier-naming.ProtectedMemberCase
+    value:  'lower_case'
+  - key:    readability-identifier-naming.ProtectedMemberSuffix
+    value:  '_'        
+  - key:    readability-identifier-naming.VariableCase
+    value:  'lower_case'
+  - key:    readability-identifier-naming.LocalVariableCase
+    value:  'lower_case'
+  - key:    readability-identifier-naming.TypeAliasCase
+    value:  'CamelCase'
+  - key:    readability-identifier-naming.UnionCase
+    value:  'CamelCase'
+  - key:    readability-identifier-naming.FunctionCase
+    value:  'camelBack'
+  - key:    readability-identifier-naming.NamespaceCase
+    value:  'lower_case'
+  - key:    readability-identifier-naming.GlobalConstantCase
+    value:  'UPPER_CASE'
+  - key:    readability-identifier-naming.EnumCase
+    value:  'CamelCase'
+  - key:    readability-identifier-naming.EnumConstantCase
+    value:  'CamelCase'
+  - key:    readability-identifier-naming.GlobalConstantPrefix
+    value:  'G_'
```

Comment:
So far I didn't look at the `.clang-tidy` files carefully and I can say that I have _no experience_ in this area (so my opinion is not strong here).

I'm not sure if the order of directives here matters, but if we can freely change the order then the following comes to my mind.
I would feel comfortable if closely related items were close together, e.g. 
`GlobalConstantCase` and `GlobalConstantPrefix`.
When I write the Global Constant, I write the prefix first, then the name. That is why I would place `GlobalConstantPrefix` first, then `GlobalConstantCase`.

If the same idea is applied to the while code writing stage, I would consider the following order:
```
NamespaceCase

AbstractClassPrefix

ClassCase
StructCase
UnionCase

FunctionCase
// Are member functions covered by `FunctionCase`?
ParameterCase

// Member types, enums, aliases/typedefs, member constants, member variables (including static). Public first, then protected, then private. Prefix first, then name.

// Non-member funcs, their params, local vars (including local statics).

// Global consts and vars (including global statics).

// Macros last (since they are not recommended, see [Item 2: Prefer consts, enums, and inlines to #defines](http://wavelino.coffeecup.com/pdf/Effective%20Cshapshap.pdf))
```

Reply:
This is a good suggestion. I've updated the `.clang-tidy` file with sections with closely related items.

Reply:
"Are member functions covered by `FunctionCase`?" <- Yes, they are, but you can specialise for member functions if you like: https://clang.llvm.org/extra/clang-tidy/checks/readability-identifier-naming.html

I've added the link to the source for future reference.